### DISCUSSION TITLE
String to Json::Value

### DISCUSSION THREAD

**Discussion by srujanareddybada** [open-source-parsers/jsoncpp#1295] [Q&A]
can i convert a string to a Json::Value using this parser - does it have any converter
For ex: 
string var;
Json::Value var.tojson() -> anything like this is possible ? or any other altrenative

**Comment by BillyDonahue** [comment]
You can just supply the `std::string` as a `Json::Value` constructor argument.

    std::string var = ...;
    Json::Value(var);

**Reply by srujanareddybada** [reply]
```c++
std::string var = "hello" ;
std::string var1 = "value1" ;
Json::Value(var);  // if done like this
Json::Value root;
```

How to add it to root value to achieve below result please suggest

    { "hello" : 
        { "value1" : 10}
    }

**Reply by BillyDonahue** [reply]
I think the simplest code to do this is probably something like:


```c++
    Json::Value root;

    Json::Value subObj;
    subObj["value1"] = 10;
    root["hello"] = subObj;
```

**Reply by srujanareddybada** [reply]
yes your right but would like to know if there is any possibility or workaround to convert a string to a Json::value (working on a specific requirement) any suggestion would be very helpful

**Comment by BillyGoto** [comment]
you pass the string to the Value constructor

On Fri, May 28, 2021 at 2:07 AM srujanareddybada ***@***.***>
wrote:

> yes your right but would like to know if there is any possibility or
> workaround to convert a string to a Json::value (working on a specific
> requirement) any suggestion would be very helpful
>
> —
>
>
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/open-source-parsers/jsoncpp/discussions/1295#discussioncomment-796360>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABYXGPDGSBAKLT3BPJK776LTP4XKRANCNFSM45UJDLYA>
> .
>
-- 
ǝnɥɐuop ʎllıq
